sdk/tests/platform/test_az_platform.c

```diff
@@ -0,0 +1,205 @@
+}
+
+#ifndef AZ_NO_PRECONDITION_CHECKING
+ENABLE_PRECONDITION_CHECK_TESTS()
+#endif // AZ_NO_PRECONDITION_CHECKING
+
+// Checking for preconditions
+#ifndef AZ_NO_PRECONDITION_CHECKING
+static void test_az_platform_clock_msec_null(void** state)
+{
+  SETUP_PRECONDITION_CHECK_TESTS();
+  (void)state;
+
+  ASSERT_PRECONDITION_CHECKED(az_platform_clock_msec(NULL));
+}
+
+static void test_az_platform_get_random_null(void** state)
+{
+  SETUP_PRECONDITION_CHECK_TESTS();
+  (void)state;
+
+  ASSERT_PRECONDITION_CHECKED(az_platform_get_random(NULL));
+}
+
+static void test_az_platform_timer_create_null(void** state)
+{
+  SETUP_PRECONDITION_CHECK_TESTS();
+  (void)state;
+
+  ASSERT_PRECONDITION_CHECKED(
+      az_platform_timer_create(NULL, _test_az_platform_timer_callback, NULL));
+}
+
+static void test_az_platform_timer_start_null(void** state)
+{
+  SETUP_PRECONDITION_CHECK_TESTS();
+  (void)state;
+
+  ASSERT_PRECONDITION_CHECKED(az_platform_timer_start(NULL, 0));
+}
+
+static void test_az_platform_timer_destroy_null(void** state)
+{
+  SETUP_PRECONDITION_CHECK_TESTS();
+  (void)state;
+
+  ASSERT_PRECONDITION_CHECKED(az_platform_timer_destroy(NULL));
+}
+
+static void test_az_platform_mutex_init_null(void** state)
+{
+  SETUP_PRECONDITION_CHECK_TESTS();
+  (void)state;
+
+  ASSERT_PRECONDITION_CHECKED(az_platform_mutex_init(NULL));
+}
+
+static void test_az_platform_mutex_acquire_null(void** state)
+{
+  SETUP_PRECONDITION_CHECK_TESTS();
+  (void)state;
+
+  ASSERT_PRECONDITION_CHECKED(az_platform_mutex_acquire(NULL));
+}
+
+static void test_az_platform_mutex_release_null(void** state)
+{
+  SETUP_PRECONDITION_CHECK_TESTS();
+  (void)state;
+
+  ASSERT_PRECONDITION_CHECKED(az_platform_mutex_release(NULL));
+}
+
+static void test_az_platform_mutex_destroy_null(void** state)
+{
+  SETUP_PRECONDITION_CHECK_TESTS();
+  (void)state;
+
+  ASSERT_PRECONDITION_CHECKED(az_platform_mutex_destroy(NULL));
+}
+
+#endif // AZ_NO_PRECONDITION_CHECKING
+
+static void test_az_platform_clock_msec_once(void** state)
+{
+  (void)state;
+
+  int64_t test_clock = 0;
+
+  assert_int_equal(az_platform_clock_msec(&test_clock), AZ_OK);
+  assert_true(test_clock >= 0);
+}
+
+static void test_az_platform_sleep_msec_one_second(void** state)
+{
+  (void)state;
+
+  int64_t test_clock_one = 0; // Time before sleep.
+  int64_t test_clock_two = 0; // Time after sleep.
+  int64_t test_expected_time = _az_TIME_MILLISECONDS_PER_SECOND;
+
+  assert_int_equal(az_platform_clock_msec(&test_clock_one), AZ_OK);
+  assert_int_equal(az_platform_sleep_msec(_az_TIME_MILLISECONDS_PER_SECOND), AZ_OK);
+  assert_int_equal(az_platform_clock_msec(&test_clock_two), AZ_OK);
+  
+  assert_true((test_clock_two - test_clock_one) >= test_expected_time);
+}
+
+static void test_az_platform_get_random_once(void** state)
+{
+  (void)state;
+
+  int32_t test_random = 0;
+
+  assert_int_equal(az_platform_get_random(&test_random), AZ_OK);
+  assert_true(test_random > 0);
```

Comment:
0 not a valid random number??

Reply:
According to the description in az_platform.h, random returns a number greater than 0. Upon reading the man page for the posix implementation of random I see that it can return 0. Updated both.